mempool/src/feerate.rs

```diff
@@ -0,0 +1,58 @@
+use common::primitives::amount::Amount;
+
+lazy_static::lazy_static! {
+    pub(crate) static ref INCREMENTAL_RELAY_FEE_RATE: FeeRate = FeeRate::new(Amount::from_atoms(1000));
+    pub(crate) static ref INCREMENTAL_RELAY_THRESHOLD: FeeRate = FeeRate::new(Amount::from_atoms(500));
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub(crate) struct FeeRate {
+    atoms_per_kb: u128,
+}
+
+impl FeeRate {
+    pub(crate) fn new(tokens_per_kb: Amount) -> Self {
+        Self {
+            atoms_per_kb: tokens_per_kb.into_atoms(),
+        }
+    }
+
+    pub(crate) fn of_tx(fee: Amount, tx_size: usize) -> Self {
+        Self {
+            atoms_per_kb: Self::div_up(1000 * fee.into_atoms(), tx_size),
+        }
+    }
+
+    pub(crate) fn compute_fee(&self, size: usize) -> Amount {
+        let size = u128::try_from(size).expect("compute_fee conversion");
+        Amount::from_atoms(self.atoms_per_kb * size / 1000)
+    }
+
+    pub(crate) fn tokens_per_kb(&self) -> u128 {
+        self.atoms_per_kb
+    }
+
+    fn div_up(fee: u128, tx_size: usize) -> u128 {
+        let tx_size = u128::try_from(tx_size).expect("div_up conversion");
+        (fee + tx_size - 1) / tx_size
+    }
+}
+
+impl std::ops::Add for FeeRate {
+    type Output = FeeRate;
+    fn add(self, other: Self) -> Self::Output {
+        let tokens_per_kb = self.atoms_per_kb + other.atoms_per_kb;
+        FeeRate {
+            atoms_per_kb: tokens_per_kb,
+        }
+    }
+}
+
+impl std::ops::Div<u128> for FeeRate {
+    type Output = FeeRate;
+    fn div(self, rhs: u128) -> Self::Output {
+        FeeRate {
+            atoms_per_kb: self.atoms_per_kb / rhs,
+        }
+    }
+}
```

Comment:
Should we use `NonZeroU128` here to make sure the division is well defined?

If this division operator is used for calculating feerate from the fee amount and transaction size, we can be even more precise with types. Introduce `TxSize` type that is known to be non-zero and have `Amount / TxSize -> FeeRate`.

No need to do this in the present PR.